eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -0,0 +1,34 @@
+parameters:
+  - name: Paths
+    type: object
+    default: []
+  - name: Repositories
+    type: object
+    default:
+      - Name: $(Build.Repository.Name)
+        Commitish: $(Build.SourceVersion)
+        WorkingDirectory: $(System.DefaultWorkingDirectory)
+
+steps:
+  - checkout: none
+
+  - ${{ each repo in parameters.Repositories }}:
+    - pwsh: |
+        $dir = "${{ coalesce(repo.WorkingDirectory, format('{0}/{1}', '$(System.DefaultWorkingDirectory)', repo.Name)) }}"
+        New-Item $dir -ItemType Directory -Force
```

Comment:
Won't the clone handle this?

Reply:
Oh it doesn't handle it because you are expecting the directory to already exist. We could probably handle this by passing the path to the clone method instead of the `.` but I'm indifferent about which approach to take.

Reply:
The problem with passing a path is then I can't the default in any parent pipeline without having to update every subsequent step with a `workingDirectory` override. I wanted to keep it simple for single-repo scenarios, since devops has the same behavior (single repo clone to `.../1/s`, multi-repo clone to `.../1/s/repo`).

Reply:
Oh I see you mean the default parameter being `.` instead of the variable.

Reply:
The problem with that is we need to construct the path to the cloned repo with expressions we can't use in the scope of the parameter definition default, and so we'd have the path reference inline to the code in multiple places, as opposed to just the one place for directory init and then as `workingDirectory` everywhere else.

Reply:
I'm good with what you have but I would expect you only need to have it inline in the clone command. All the other git steps can be independent and have the working directory set.